hc/api/tests/test_ping.py

```diff
@@ -44,22 +39,51 @@ def test_it_handles_120_char_ua(self):
               "AppleWebKit/537.36 (KHTML, like Gecko) "
               "Chrome/44.0.2403.89 Safari/537.36")
 
-        check = Check()
-        check.save()
-        r = self.client.get("/ping/%s/" % check.code, HTTP_USER_AGENT=ua)
+        r = self.client.get("/ping/%s/" % self.check.code, HTTP_USER_AGENT=ua)
         assert r.status_code == 200
 
-        pings = list(Ping.objects.all())
-        assert pings[0].ua == ua
+        ping = Ping.objects.latest("id")
+        assert ping.ua == ua
 
     def test_it_truncates_long_ua(self):
         ua = "01234567890" * 30
 
-        check = Check()
-        check.save()
-        r = self.client.get("/ping/%s/" % check.code, HTTP_USER_AGENT=ua)
+        r = self.client.get("/ping/%s/" % self.check.code, HTTP_USER_AGENT=ua)
+        assert r.status_code == 200
+
+        ping = Ping.objects.latest("id")
+        assert len(ping.ua) == 200
+        assert ua.startswith(ping.ua)
+
+    def test_it_reads_forwarded_ip(self):
+        ip = "1.1.1.1"
+        r = self.client.get("/ping/%s/" % self.check.code,
+                            HTTP_X_FORWARDED_FOR=ip)
+        ping = Ping.objects.latest("id")
+        assert r.status_code == 200
+        assert ping.remote_addr == "1.1.1.1"
+
+        ip = "1.1.1.1, 2.2.2.2"
```

Comment:
if the incoming request passed through **multiple** proxies on its way, this will be a list of IPs and the first one is the originating source